### PR TITLE
Replace appweb with httplib

### PR DESCRIPTION
**Pull request by nikodemak**

### CODE REVIEW

File: src/lobby/lobbyapp.cpp [hunk 1]
```diff
@@ -735,66 +735,38 @@ bool CLobbyApp::GetRankForCallsign(const char* szPlayerName, int *rank, double *
 {
```
Comment (nikodemak) [inline src/lobby/lobbyapp.cpp]: is this function still being used? isn't ACSS gone?

Comment (BackTrak) [reply src/lobby/lobbyapp.cpp]: ACSS is gone, and I just checked the server, that registry for the lobby points to the old ACSS server and is for sure not going anywhere. So, you can remove this. Thanks!

**Comment by nikodemak** [comment]
well I'm not exactly sure how to set up a lobby server (for testing the chat logging function in lobby) or how to change all of the older/newer VS dependencies but otherwise this is ready to be merged

**Comment by BackTrak** [comment]
Hi Niko, you can find local lobby / local server instructions by installing the Allegiance Server from steam. See the readme_server.txt file in there. 

I will cut/paste here for you:


---------

Hi there!

Allegiance is an old game, and so it will need to have DPlay installed for it's communications. 
Run the dplay_install.bat if you don't already have it installed. 

Next, run server_install.bat to setup the Allegiance Server windows service. 

Then, edit the values in this registry path to your liking: 
HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Microsoft Games\Allegiance\1.4\Server

Finally, you should be able to start the Allegiance Windows Service. 


You can also edit Artwork\Cores.txt to turn cores on and off, and you can edit Artwork\Maps.txt to turn maps on and off. 

If you want your stats to be persistant, contact BackTrak for an API key and to have your server added to the official servers list.

Running AllSrv.exe -RegServer will put the server into interactive mode. You can then run it from the command line 
with out any other arguments. This can help with debugging a stuck server installation. Run AllSrv -Service to return it
to Windows Service mode.

------- If you want to run a server environment for debugging (These steps are normally only done by a developer) -----------

If you want to setup a local environment for testing, or for LAN play, the steps are a little different. You will need to 
setup your Allegiance client registry to point to a special config file so that it will know to talk to your local lobby
instead of the public one. Building a FZDebug server will also enable you to use all of the cheat codes which make local 
testing a lot easier. Once you have a local server environment running, it is then very easy to launch the lobby and 
server using the debugger. 

1. Change HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Microsoft Games\Allegiance\1.4\Server\CfgFile from http://allegiance.zaphop.com/allegiance-server.txt to http://allegiance.zaphop.com/allegiance-local.txt
2. If you are running a local Allegiance client on the same machine, modify HKEY_CURRENT_USER\Software\Microsoft\Microsoft Games\Allegiance\1.4\CfgFile from http://allegiance.zaphop.com/allegiance.txt to http://allegiance.zaphop.com/allegiance-local.txt This will point your Allegiance client to your local lobby and server.
2. Start the lobby (c:\Allegiance\Server\AllLobby.exe) 
3. Wait until the Lobby says that it's Ready for Clients To Connect (This is VERY important)
4. Start AllSrv

You can then connect your local Allegiance client to your Local Lobby and Local Server.

**Review by BackTrak — APPROVED** [review]
Thanks for the cleanup!

**Comment by BackTrak** [comment]
Crash on game end when posting chat back to lobby web service, I had to put this bit back:

		postData += "&sourceSteamID=" + pfmLogChatMessage->sourceSteamID;

		postData += "&targetSteamID=" + pfmLogChatMessage->targetSteamID;



To: 

		char steamID[256];
		sprintf(steamID, "%" PRIu64, pfmLogChatMessage->sourceSteamID);
		postData += "&sourceSteamID=" + std::string(steamID);

		sprintf(steamID, "%" PRIu64, pfmLogChatMessage->targetSteamID);
		postData += "&targetSteamID=" + std::string(steamID);

Otherwise, seemed to work ok!

**Comment by BackTrak** [comment]
Released to servers now.

**Comment by BackTrak** [comment]
Fixed another bug, std::string can't take a NULL input, and message can be NULL in the chat. Added the NULL check to a new template FM_VAR_REF_STDSTRING.

                std::string missionName = FM_VAR_REF(pfmLogChatMessage, MissionName);
		std::string sourceName = FM_VAR_REF(pfmLogChatMessage, SourceName);
		std::string targetName = FM_VAR_REF(pfmLogChatMessage, TargetName);
		std::string message = FM_VAR_REF(pfmLogChatMessage, Message);
		std::string sourceIP = FM_VAR_REF(pfmLogChatMessage, SourceIP);

to:

		std::string missionName = FM_VAR_REF_STDSTRING(pfmLogChatMessage, MissionName);
		std::string sourceName = FM_VAR_REF_STDSTRING(pfmLogChatMessage, SourceName);
		std::string targetName = FM_VAR_REF_STDSTRING(pfmLogChatMessage, TargetName);
		std::string message = FM_VAR_REF_STDSTRING(pfmLogChatMessage, Message);
		std::string sourceIP = FM_VAR_REF_STDSTRING(pfmLogChatMessage, SourceIP);

**Pull request merged**